chain/indexer/src/streamer/senders.rs

```diff
@@ -0,0 +1,207 @@
+use std::collections::HashMap;
+
+use futures::StreamExt;
+use near_async::messaging::{AsyncSender, SendAsync};
+use near_chain_configs::ProtocolConfigView;
+use near_client::{
+    GetBlock, GetChunk, GetExecutionOutcomesForBlock, GetProtocolConfig, GetReceipt,
+    GetStateChangesWithCauseInBlockForTrackedShards, Status, StatusResponse,
+};
+use near_client_primitives::types::{
+    GetBlockError, GetChunkError, GetProtocolConfigError, GetReceiptError, GetStateChangesError,
+    StatusError,
+};
+use near_epoch_manager::shard_tracker::ShardTracker;
+use near_indexer_primitives::IndexerExecutionOutcomeWithOptionalReceipt;
+use near_o11y::span_wrapped_msg::{SpanWrapped, SpanWrappedMessageExt};
+use near_primitives::hash::CryptoHash;
+use near_primitives::types::{BlockId, BlockReference, ShardId};
+use near_primitives::views::{
+    BlockView, ChunkView, ExecutionOutcomeWithIdView, ReceiptView, StateChangesView,
+};
+use tracing::warn;
+
+use crate::INDEXER;
+use crate::streamer::errors::FailedToFetchData;
+
+#[derive(Clone, near_async::MultiSend, near_async::MultiSenderFrom)]
+pub struct IndexerViewClientSender {
+    pub get_block_sender: AsyncSender<GetBlock, Result<BlockView, GetBlockError>>,
+    pub get_chunk_sender: AsyncSender<GetChunk, Result<ChunkView, GetChunkError>>,
+    pub get_protocol_config_sender:
+        AsyncSender<GetProtocolConfig, Result<ProtocolConfigView, GetProtocolConfigError>>,
+    pub get_execution_outcomes_for_block_sender: AsyncSender<
+        GetExecutionOutcomesForBlock,
+        Result<HashMap<ShardId, Vec<ExecutionOutcomeWithIdView>>, String>,
+    >,
+    pub get_receipt_sender: AsyncSender<GetReceipt, Result<Option<ReceiptView>, GetReceiptError>>,
+    pub get_state_changes_with_cause_in_block_for_tracked_shards_sender: AsyncSender<
+        GetStateChangesWithCauseInBlockForTrackedShards,
+        Result<HashMap<ShardId, StateChangesView>, GetStateChangesError>,
+    >,
+}
+
+#[derive(Clone, near_async::MultiSend, near_async::MultiSenderFrom)]
+pub struct IndexerClientSender {
+    pub status_sender: AsyncSender<SpanWrapped<Status>, Result<StatusResponse, StatusError>>,
+}
+
+impl IndexerViewClientSender {
```

Comment:
It's a bit unnatural for a Sender to have an impl like this but I'm guessing you kept this for minimal code changes? Might make sense to add a quick comment above the impl block saying this isn't the typical way to use Senders/MultiSenders and that ideally we should be calling the send and send_async functions directly.

Reply:
That is a good point, thanks for flagging!
I've added separate `*Fetcher` structs wrapping senders to avoid that: 369f7adfa.